src/bp/core/routers/auth.ts

```diff
@@ -27,131 +29,132 @@ export class AuthRouter extends CustomRouter {
-    const role = await this.workspaceService.getRoleForUser(tokenUser!.email)
-    if (!role) {
-      throw new NotFoundError(`Role for user "${tokenUser!.email}" doesn't exist`)
-    }
-    return sendSuccess(res, "Retrieved user's permissions successfully", role.rules)
-  }
+    router.get(
+      '/me/profile',
+      this.checkTokenHeader,
+      this.asyncMiddleware(async (req: RequestWithUser, res) => {
+        const { email, strategy, isSuperAdmin } = req.tokenUser!
 
-  sendSuccess = async (req, res) => {
-    return sendSuccess(res)
-  }
+        const user = await this.authService.findUser(email, strategy)
+        if (!user) {
+          throw new NotFoundError(`User ${email || ''} not found`)
+        }
+        const { firstname, lastname } = user.attributes
 
-  async setupRoutes() {
-    const router = this.router
+        const userRole = await this.workspaceService.getRoleForUser(email, strategy, req.workspace!)
 
-    const authStrategy = await this.getAuthStrategy()
+        const userProfile = {
+          ...user,
+          email,
+          isSuperAdmin,
+          fullName: [firstname, lastname].filter(Boolean).join(' '),
+          permissions: userRole && userRole.rules
+        }
 
-    if (process.IS_PRO_ENABLED && authStrategy.strategy !== 'basic') {
-      this.authStrategies.setup(router)
-    } else {
-      router.post('/login', this.asyncMiddleware(this.login))
-      router.post('/register', this.asyncMiddleware(this.register))
-    }
+        return sendSuccess(res, 'Retrieved profile successfully', userProfile)
+      })
+    )
 
-    router.get('/config', this.asyncMiddleware(this.sendConfig))
+    router.post(
+      '/me/profile',
+      this.checkTokenHeader,
+      this.asyncMiddleware(async (req: RequestWithUser, res) => {
+        const { email, strategy } = req.tokenUser!
+
+        validateBodySchema(
+          req,
+          Joi.object().keys({
+            firstname: Joi.string()
+              .min(0)
+              .max(35)
+              .trim()
+              .required(),
+            lastname: Joi.string()
+              .min(0)
+              .max(35)
+              .trim()
+              .required()
+          })
+        )
+
+        await this.authService.updateAttributes(email, strategy, {
+          firstname: req.body.firstname,
+          lastname: req.body.lastname
+        })
+
+        return sendSuccess(res, 'Updated profile successfully')
+      })
+    )
 
-    router.get('/ping', this.checkTokenHeader, this.asyncMiddleware(this.sendSuccess))
+    router.get(
+      '/me/workspaces',
+      this.checkTokenHeader,
+      this.asyncMiddleware(async (req: RequestWithUser, res) => {
+        const { email, strategy, isSuperAdmin } = req.tokenUser!
 
-    router.get('/me/profile', this.checkTokenHeader, this.asyncMiddleware(this.getProfile))
+        if (!isSuperAdmin) {
+          return res.send(await this.workspaceService.getUserWorkspaces(email, strategy))
```

Comment:
try not to send top level json arrays, it encloses a known security flaw. 
see here . : https://haacked.com/archive/2008/11/20/anatomy-of-a-subtle-json-vulnerability.aspx/

Reply:
I'm taking a good note of that vulnerability ! (I do get the reason for sendSuccess haha..) However, there's no sensitive information returned from that profile, all public information, and no information returned here can be used for anything else.